### PR TITLE
dist/tools/buildsystem_sanity_check: minor fixes and annotate errors in Github Action

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This provides Github annotation support for the `buildsystem_sanity_check` check.

### Testing procedure
I provided a test commit that will cause some annotations.

### Issues/PRs references
None

### CODE REVIEW

File: sys/net/gnrc/network_layer/ipv6/Makefile [hunk 1]
```diff
@@ -1,3 +1,7 @@
 MODULE = gnrc_ipv6
 
+ifeq (,$(filter FEATURES_PROVIDED,periph_gpio))
+  # this should not be
+endif
```
Comment (miri64) [inline sys/net/gnrc/network_layer/ipv6/Makefile:5]: These are reported in the `static-tests` output as annotations, but there seems to be a cap of 10 per check, for if they are reported in the summary or in the file.

**Comment by jia200x** [comment]
well, it seems to work!

**Comment by miri64** [comment]
Shall I remove the test commit?

**Review by jia200x — APPROVED** [review]
ACK. Please remove the bogus commit

**Comment by miri64** [comment]
Done.

**Comment by miri64** [comment]
Thanks for the review!

**Pull request merged**